core/src/test/java/org/apache/iceberg/TestMetrics.java

```diff
@@ -95,59 +88,57 @@
       optional(8, "dateCol", DateType.get()),
       required(9, "timeCol", TimeType.get()),
       required(10, "timestampCol", TimestampType.withoutZone()),
-      optional(11, "uuidCol", UUIDType.get()),
-      required(12, "fixedCol", FixedType.ofLength(4)),
-      required(13, "binaryCol", BinaryType.get())
+      required(11, "fixedCol", FixedType.ofLength(4)),
+      required(12, "binaryCol", BinaryType.get())
   );
 
-  private final UUID uuid = UUID.randomUUID();
-  private final GenericFixed fixed = new GenericData.Fixed(
-      org.apache.avro.Schema.createFixed("fixedCol", null, null, 4),
-      "abcd".getBytes(StandardCharsets.UTF_8));
+  private final byte[] fixed = "abcd".getBytes(StandardCharsets.UTF_8);
 
   public abstract Metrics getMetrics(InputFile file);
 
-  public abstract File writeRecords(Schema schema, Record... records) throws IOException;
+  public abstract InputFile writeRecords(Schema schema, Record... records) throws IOException;
 
-  public abstract File writeRecords(Schema schema, Map<String, String> properties, GenericData.Record... records)
+  public abstract InputFile writeRecordsWithSmallRowGroups(Schema schema, Record... records)
       throws IOException;
 
-  public abstract int splitCount(File parquetFile) throws IOException;
+  public abstract int splitCount(InputFile parquetFile) throws IOException;
+
+  public boolean supportsSmallRowGroups() {
+    return false;
+  }
 
   @Test
   public void testMetricsForRepeatedValues() throws IOException {
```

Comment:
is the test name misleading? Seems like we. are testing for multiple records, not repeated values in maps and lists

Reply:
I think it sounds like we were testing repeated values in maps and lists.

Reply:
Since this is an existing test name, I'd rather not update it in this PR. We can open a separate one to fix it if you think it is necessary.